datafusion/physical-expr/src/sort_expr.rs

```diff
@@ -116,9 +116,11 @@ impl PhysicalSortRequirement {
     }
 }
 
-pub fn make_sort_requirements_from_exprs(
-    ordering: &[PhysicalSortExpr],
-) -> Vec<PhysicalSortRequirement> {
+pub type ExprOrdering = Vec<PhysicalSortExpr>;
+pub type ExprOrderingRef<'a> = &'a [PhysicalSortExpr];
+pub type OrderingRequirement = Vec<PhysicalSortRequirement>;
```

Comment:
While these do simplify the signatures, I am a little worried about just adding a type signature without additional supporting documentation.

If we are going to change the signatures again what do you think about going all the way and actually making structures

```rust
struct ExprOrdering {
  exprs: Vec<PhysicalSortExpr>;
}
```

And then make functions like (with documentation):

```rust
impl ExprOrdering {
  fn new (exprs: Vec<PhysicalSortExpr>) -> Self {.._

  fn from_requirements(requirements: OrderingRequirement -> Self {..}
}
```

For example, here is a PR that tries to encapsulate more of the `PhysicalSortRequirements` along with a bunch of docs: https://github.com/apache/arrow-datafusion/pull/5863

Reply:
I tried a refactor using full-fledged types but the resulting changes were more involved/time-consuming, so I opted to go with aliases until we know there are concrete use cases requiring multiple attributes in types such as `ExprOrdering`.

I see two options:
- We can just merge #5863 and defer/revisit this kind of a refactor when we have a real use case for multiple fields in types such as `ExprOrdering`.
- We can merge both and do another PR when/if such use cases arise.

I am OK with both, up to you.

Reply:
> I tried a refactor using full-fledged types but the resulting changes were more involved/time-consuming, so I opted to go with aliases until we know there are concrete use cases requiring multiple attributes in types such as ExprOrdering.


This makes sense -- thank you for trying. It definitely took me far more time to make https://github.com/apache/arrow-datafusion/pull/5863 than I expected -- mostly to come up with coherent description of what the structures did.

> We can just merge https://github.com/apache/arrow-datafusion/pull/5863 and defer/revisit this kind of a refactor when we have a real use case for multiple fields in types such as ExprOrdering.

I think this is my preference for the moment. Let's wait to see if anyone else has comments

Reply:
@mingmwang, what do you think? Any preference on your side?

Reply:
I will close this for now, we can revisit this issue in the future if we like.